src/components/Api.js

```diff
@@ -0,0 +1,87 @@
+export default class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  _request(url, options) {
+    return fetch(`${url}${options}`, { headers: this._headers }).then(
```

Comment:
`_request` should look like that to be able to use it everywhere

```js
  _request(url, options) {
    return fetch(url, options).then(this._checkPromise);
  }

```


And then use it


```js

  _request(url, options) {
    return fetch(url, options).then(this._checkPromise);
  }

  _checkPromise(res) {
    if (res.ok) {
      return res.json();
    }
    // if the server returns an error, reject the promise
    return Promise.reject(`Error: ${res.status}`);
  }

  getInitialCards() {
    return this._request(`${this._baseURL}/cards`, { headers: this._headers });
  }

  getUserInfo() {
    return this._request(`${this._baseURL}/users/me`, {
      headers: this._headers,
    });
  }
```

Reply:
this setup of _request works for the shown methods within Api.js, but if i try to implement it for updateUserInfo(name, about) or any other none GET request it does not work...do I need to refactor _request(url, options) in order to use _request for all the requests? Or am I looking in the wrong place to fix this error?